### PR TITLE
Service Declaration - Service Line Use Correct Header

### LINKED ISSUE #27462: [Bug]: Wrong Header.Get(...,...) in Service Declaration - Service Line
### Describe the issue

The code in the OnValidate Trigger of the Service Line tries to get a SalesHeader instead of the ServiceHeader
https://github.com/microsoft/ALAppExtensions/blob/main/Apps/W1/ServiceDeclaration/app/src/ServiceDocuments/ServDeclServLine.TableExt.al

### Expected behavior

The ServiceHeader should be used.

### Steps to reproduce

Create a Service Header
Create a Service Line
Validate the field "Applicable For Serv. Decl."

### Additional context

_No response_

### I will provide a fix for a bug

- [x] I will provide a fix for a bug

### PR DESCRIPTION
**Pull request by pri-kise**
#### Summary
Use correct Header

#### Work Item(s)
Fixes #27462 
Fixes [AB#555792](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/555792)

### CODE REVIEW

**Review by mjmatthiesen — APPROVED** [review]

File: Apps/W1/ServiceDeclaration/app/src/ServiceDocuments/ServDeclServLine.TableExt.al [hunk 1]
```diff
@@ -34,11 +33,11 @@ tableextension 5034 "Serv. Decl. Serv. Line" extends "Service Line"
 
             trigger OnValidate()
             var
-                SalesHeader: Record "Sales Header";
+                ServiceHeader: Record "Service Header";
             begin
-                SalesHeader.Get("Document Type", "Document No.");
+                ServiceHeader := Rec.GetServHeader();
```
Comment (tino-teuber) [inline Apps/W1/ServiceDeclaration/app/src/ServiceDocuments/ServDeclServLine.TableExt.al:38]: It is better to check first whether `Applicable For Serv. Decl.` is `true` before requesting the service header? Like this:
```al
begin
    if "Applicable For Serv. Decl." then begin
        ServiceHeader := Rec.GetServHeader();
        ServiceHeader.TestField("Applicable For Serv. Decl.");
    end:
end;
```

**Review by JesperSchulz — APPROVED** [review]

File: Apps/W1/ServiceDeclaration/app/src/ServiceDocuments/ServDeclServLine.TableExt.al (hunk 1 above)
Comment (JesperSchulz) [reply Apps/W1/ServiceDeclaration/app/src/ServiceDocuments/ServDeclServLine.TableExt.al:38]: That is true. I've processed the PR "as is", but will make that change in a follow-up PR.

**Review by JesperSchulz — APPROVED** [review]

**Comment by JesperSchulz** [comment]
Thanks for reporting this. We agree, and we’ll publish a fix asap, either in an update for the current version or in the next major release. Please do not reply to this, as we do not monitor closed issues. If you have follow-up questions or requests, please create a new issue where you reference this one.

Build ID: 25897.

**Comment by JesperSchulz** [comment]
> This bug exists still in extension "Service Declaration" 25.3 > in tableextension 5034 "Serv. Decl. Serv. Line" (extends "Service Line"): In OnValidate of field 5011 "Applicable For Serv. Decl." there is a line of code that does the following: SalesHeader.Get("Document Type", "Document No."). That gives us an error because we're on a service line, not on a sales line... When will this be up-to-date?

If nothing else is specified, fixes from GitHub go into the next major release. As this was merged in October, the next major release is the April one, being 2025 wave 1 aka. v26.

**Pull request merged**